### PR TITLE
Group get closest plot data

### PR DESCRIPTION
**Pull request by istarnx**
Note that PlotData now includes "plot." I updated the code in a number of places to include this data.

### CODE REVIEW

File: src/components/componentGroup.ts [hunk 1]
```diff
@@ -23,6 +23,48 @@ export module Component {
...
+     * @param {Point} queryPoint The point to which plot data should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): Plot.PlotData {
+      var minDistSquared = Infinity;
+      var closestPlotData: Plot.PlotData = {
+        data: [],
+        pixelPoints: [],
+        plot: null,
```
Comment (bluong) [inline src/components/componentGroup.ts]: Is this plot parameter valid?  Imagine if I were to merge a scatterPlot and a barPlot

Comment (bluong) [inline src/components/componentGroup.ts]: Oh so closestPlotData only returns a singular item?  Is this not consistent with what plot.getClosestPlotData does?

File: src/components/componentGroup.ts [hunk 2]
```diff
@@ -23,6 +23,48 @@ export module Component {
       components.forEach((c: AbstractComponent) => this._addComponent(c));
     }
 
+    /**
+     * Retrieves closest PlotData to queryPoint across all plots in this group.
+     * Plots define their own notion of closeness; this function compares results across
```
Comment (bluong) [inline src/components/componentGroup.ts]: How does the function compare results / what are the results you are referring to?

File: src/components/componentGroup.ts [hunk 3]
```diff
@@ -23,6 +23,48 @@ export module Component {
...
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): Plot.PlotData {
+      var minDistSquared = Infinity;
+      var closestPlotData: Plot.PlotData = {
+        data: [],
+        pixelPoints: [],
+        plot: null,
+        selection: d3.selectAll([])
```
Comment (bluong) [inline src/components/componentGroup.ts]: Use d3.select() instead.  It'll give you a null selection

File: src/components/componentGroup.ts [hunk 4]
```diff
@@ -23,6 +23,48 @@ export module Component {
...
+      };
+
+      this.components().forEach((c) => {
+        // consider only components implementing this function
+        if ((<any>c).getClosestPlotData !== undefined) {
```
Comment (bluong) [inline src/components/componentGroup.ts]: I personally like to put a space between the cast and the variable like
(<any> c)

Comment (bluong) [inline src/components/componentGroup.ts]: We should not be anycasting this.  You should be able to do a call like...

if (c instanceof Plot.AbstractPlot)

I think

File: src/components/componentGroup.ts [hunk 5]
```diff
@@ -23,6 +23,48 @@ export module Component {
...
+          var cpd = (<any>c).getClosestPlotData(queryPoint);
```
Comment (bluong) [inline src/components/componentGroup.ts]: Cast c to Plot.AbstractPlot

File: src/components/componentGroup.ts [hunk 6]
```diff
@@ -23,6 +23,48 @@ export module Component {
...
+          if (cpd.data.length > 0) {
```
Comment (bluong) [inline src/components/componentGroup.ts]: If all we're caring about is the pixelPoint, let's go with the more semantical appraoch

var plotClosestPixelPoint (or so) = cpd.pixelPoints[0]
if (plotClosestPixelPoint != null) {
do stuff
}

File: test/core/componentGroupTests.ts [hunk 7]
```diff
@@ -284,4 +284,67 @@ describe("ComponentGroups", () => {
...
+        selection: c2Selection
+      };
+
+      var c1 = new Plottable.Component.AbstractComponent();
+      (<any>c1).getClosestPlotData = (p: Plottable.Point) => {
+        return c1PlotData;
+      };
+
+      var c2 = new Plottable.Component.AbstractComponent();
+      (<any>c2).getClosestPlotData = (p: Plottable.Point) => {
```
Comment (bluong) [inline test/core/componentGroupTests.ts]: Don't anycast if you can help it

**Comment by istarnx** [comment]
Closing to re-target at develop: https://github.com/palantir/plottable/pull/1877

**Pull request closed without merging**